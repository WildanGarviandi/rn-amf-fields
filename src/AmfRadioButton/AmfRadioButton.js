import React, { Component } from 'react'
import fieldPropTypes from '../fieldPropTypes'
import PropTypes from 'prop-types'
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native'
import RadioForm, { RadioButton, RadioButtonInput, RadioButtonLabel } from 'react-native-simple-radio-button'

class AmfRadioButton extends Component {
  constructor(props) {
    super(props)

    this.checkError()
  }

  checkError() {
    const { label, options, value, onChange } = this.props

    if (!label) {
      throw new TypeError('Label for AmfRadioButton not exists. Did you forget to add label')
    }

    if (!options) {
      throw new TypeError('Field "' + label + '" (AmfRadioButton) doesnt have options')
    }
  }

  render() {
    const { label, options, value, onChange, error } = this.props

    let validationText
    let selected = options.find(i => i.value === value)
    if (error !== '') validationText = <Text style={style.errorText}>{error}</Text>

    return (
      <View>
        <Text style={style.label}>{this.props.label}</Text>
        <RadioForm formHorizontal={this.props.layout === 'horizontal' } style={{alignItems: 'flex-start'}}>
          { options.map( (option, i) => (
            <RadioButton labelHorizontal={true} key={i}>
              {/*  You can set RadioButtonLabel before RadioButtonInput */}
              <RadioButtonInput
                obj={option}
                index={i}
                isSelected={option.value === value}
                onPress={onChange}
                borderWidth={1}
                buttonInnerColor={'#009688'}
                buttonOuterColor={option.value === value ? '#009688' : '#000000' }
                buttonSize={16}
                buttonOuterSize={24}
                buttonStyle={{}}
                buttonWrapStyle={{marginLeft: 10}}
              />
              <RadioButtonLabel
                obj={option}
                index={i}
                labelHorizontal={true}
                onPress={onChange}
                labelStyle={{}}
                labelWrapStyle={{}}
              />
            </RadioButton>
          ))}

        </RadioForm>
        { validationText }
      </View>
    )
  }
}

const style = StyleSheet.create({
  button: {
    backgroundColor: 'white',
    elevation: 4,
    padding: 8,
    margin: 4,
    minWidth: 80,
    borderRadius: 32
  },
  buttonText: {
    textAlign: 'center'
  },
  buttonSelected: {
    backgroundColor: '#7ED321',
    elevation: 4,
    padding: 8,
    margin: 4,
    minWidth: 80,
    borderRadius: 32
  },
  buttonTextSelected: {
    textAlign: 'center',
    color: 'white'
  },
  errorText: {
    color: 'red',
    margin: 4,
    marginBottom: 0
  },
  label: {
    color: '#454545',
    marginRight: 16,
    marginBottom: 8
  },
  radioForm: {
    padding: 8
  }
})

AmfRadioButton.propTypes = {
  ...fieldPropTypes,
  layout: PropTypes.oneOf(['horizontal', 'vertical']),
}

export default AmfRadioButton
