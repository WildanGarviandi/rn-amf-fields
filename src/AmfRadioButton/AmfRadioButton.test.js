import React from 'react'
import AmfRadioButton from './AmfRadioButton'
import renderer from 'react-test-renderer'

const label = 'Label'
const items = [{
  label: 'foo',
  value: 'foo'
}, {
  label: 'bar',
  value: 'bar'
}]
const onPress = jest.fn()

test('AmfRadioButton simple render', () => {
  let props = { label, items, onPress }
	props.options = [
		{ label: 'Label1', value: 'value2' },
		{ label: 'Label2', value: 'value3' },
	]

  const component = renderer.create(<AmfRadioButton {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test.skip('AmfRadioButton with validation', () => {
  let props = { label, items, onPress,
    required: true,
    value: ''
  }

  const component = renderer.create(<AmfRadioButton {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

/**
 *
 */
test('AmfRadioButton without options', () => {
  let props = { label, items, onPress }

	console.error = jest.fn();

	expect(() => {
		const component = renderer.create(<AmfRadioButton {...props} />)
		let tree = component.toJSON()
	}).toThrowErrorMatchingSnapshot();
})
