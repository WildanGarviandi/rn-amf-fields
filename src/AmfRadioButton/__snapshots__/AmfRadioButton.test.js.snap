// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfRadioButton simple render 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
        "marginRight": 16,
      }
    }
  >
    Label
  </Text>
  <View
    style={
      Array [
        undefined,
        Object {
          "alignItems": "flex-start",
        },
        false,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#000000",
                "borderRadius": 12,
                "borderWidth": 1,
                "height": 24,
                "justifyContent": "center",
                "opacity": 1,
                "width": 24,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {},
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": undefined,
                },
                Object {},
              ]
            }
          >
            Label1
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#000000",
                "borderRadius": 12,
                "borderWidth": 1,
                "height": 24,
                "justifyContent": "center",
                "opacity": 1,
                "width": 24,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {},
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": undefined,
                },
                Object {},
              ]
            }
          >
            Label2
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  />
</View>
`;

exports[`AmfRadioButton with validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
        "marginRight": 16,
      }
    }
  >
    Label
  </Text>
  <View
    style={
      Array [
        undefined,
        Array [
          Object {
            "padding": 8,
          },
          Object {
            "borderColor": "red",
            "borderWidth": 1,
          },
        ],
        false,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="radioButtonInput0"
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "black",
                "borderRadius": 11,
                "borderWidth": 3,
                "height": 22,
                "justifyContent": "center",
                "opacity": 1,
                "width": 22,
              }
            }
            testID="radioButtonInputradioButton|0"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="radioButtonLabel0"
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              undefined,
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID="radioButtonLabelradioButton|0"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": "#000",
                },
                Object {
                  "marginBottom": 6,
                },
              ]
            }
          >
            foo
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="radioButtonInput1"
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "black",
                "borderRadius": 11,
                "borderWidth": 3,
                "height": 22,
                "justifyContent": "center",
                "opacity": 1,
                "width": 22,
              }
            }
            testID="radioButtonInputradioButton|1"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="radioButtonLabel1"
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              undefined,
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID="radioButtonLabelradioButton|1"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": "#000",
                },
                Object {
                  "marginBottom": 6,
                },
              ]
            }
          >
            bar
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label
     harus dipilih
  </Text>
</View>
`;

exports[`AmfRadioButton without options 1`] = `"Field \\"Label\\" (AmfRadioButton) doesnt have options"`;
